### PR TITLE
Feat/add grantee account support

### PR DESCRIPTION
**Pull request by aarmoa**
- Refactored the liquidation bot logic to allow the use of delegated (granter/grantee) accounts to submit the position liquidation message
- Updated the README.md file to clarify the configuration differences to run the bot with a normal account or with a delegated account
- Added pre-commit validations to the project
- Added two GitHub workflows to
    - Run all pre-commit validations
    - Run all unit tests with coverage and upload results to CodeCov

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,4 +64,25 @@ The following configuration options can be specified for the bot to use a Cosmos
...
\ No newline at end of file
+- LIQUIDATOR_COSMOS_USE_LEDGER
+
+#### Configuration to run the bot using a grantee account
+The bot can also be configured to run using a delegated account. With this configuration the bot will use two accounts: the granter and the grantee.
+
+- Granter account: is the account that has balances to create the liquidation order and will be associated to the position created once the liquidation order is executed
+- Grantee account: it the account used to sign the transactions broadcasted by the bot. The only requirement for the account is to have enough INJ tokens to pay for the gas fee for the transactions
+
+When running using a delegated account, the bot will send the `MsgLiquidatePosition` with the `authz` module, including it in a `MsgExecute` message,
```
Comment (nicolasbaum) [inline README.md]: When running using a delegated account, message `MsgLiquidatePosition` is not sent directly, but wrapped inside `MsgExecute` which is executed through `authz` module's authorization mechanism.

File: README.md [hunk 2]
```diff
@@ -64,4 +64,25 @@ The following configuration options can be specified for the bot to use a Cosmos
...
+
+To activate the delegated account mode the user has to configure the following options:
+
+- LIQUIDATOR_GRANTER_PUBLIC_ADDRESS: public Injective address of the account to be used as granter account
+- LIQUIDATOR_GRANTER_SUBACCOUNT_INDEX: index number of the subaccount from LIQUIDATOR_GRANTER_PUBLIC_ADDRESS to be used
+
+To deactivate the delegated account mode the LIQUIDATOR_GRANTER_PUBLIC_ADDRESS option has to be left empty.
```
Comment (nicolasbaum) [inline README.md]: I would clarify empty like LIQUIDATOR_GRANTER_PUBLIC_ADDRESS="" or empty like LIQUIDATOR_GRANTER_PUBLIC_ADDRESS=

Comment (aarmoa) [reply README.md]: Changed

File: README.md (hunk 1 above)
Comment (aarmoa) [reply README.md]: Changed

**Review by nicolasbaum — APPROVED** [review]
Addition of granter/grantee and GitHub workflows look ok to me

**Pull request merged**